engine/common/rpc/convert/convert.go

```diff
@@ -393,3 +394,68 @@ func BytesToInmemSnapshot(bytes []byte) (*inmem.Snapshot, error) {
 
 	return inmem.SnapshotFromEncodable(encodable), nil
 }
+
+func ProtoToExecutionResult(proto *entities.ExecutionResult) (*flow.ExecutionResult, error) {
+	// convert Chunks
+	parsedChunks := make(flow.ChunkList, len(proto.Chunks))
```

Comment:
nit: Add parsedChunks and parsedServiceEvents parsing logic into their own funcs